JWT, MAPPER, USER CRUD

MP-David opened the pull request:
- Estrutura arquitetura de packges
- Criação do Spring SecurityConfig
- Criação do JwtAuthenticationFilter
- Crud User
- Adicionado Mapper DTO
- Adicionado Properties de banco de JWT

MP-David committed:
Jwt Config

- Implementado libs do security checkin, jwt
- Criado User base
- Criado SecurityConfig base
- Criado JwtService e Filter base
- Adequado packges

MP-David committed:
Merge branch 'develop' into feature/config-jwt

MP-David committed:
- Estrutura arquitetura de packges
- Criação do Spring SecurityConfig
- Criação do JwtAuthenticationFilter
- Crud User
- Adicionado Mapper DTO
- Adicionado Properties de banco de JWT

MP-David committed:
- Estrutura arquitetura de packges
- Criação do Spring SecurityConfig
- Criação do JwtAuthenticationFilter
- Adicionado Mapper DTO
- Adicionado Properties de banco de JWT

MP-David committed:
- Crud User

MP-David committed:
Fix DTO, Response

CamiBregalda commented on backend/src/main/java/com/utfpr/donare/controller/UserController.java:
```diff
@@ -0,0 +1,47 @@
+
+    private final UserService userService;
+
+    @PostMapping(path = "/create")
```
Corrija a rota:
@PostMapping(value = "", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)

PS. Adiciona as anotações do swagger padrões como @Operation, @ApiResponses,

CamiBregalda commented on backend/src/main/java/com/utfpr/donare/controller/UserController.java:
```diff
@@ -0,0 +1,47 @@
+import java.util.List;
+
+@RequiredArgsConstructor
+@RequestMapping(path = "user")
```
Corrija o requestMapping para "usuarios"

CamiBregalda commented on backend/src/main/java/com/utfpr/donare/controller/UserController.java:
```diff
@@ -0,0 +1,47 @@
+        return userService.autenticar(userRequestDTO.getEmail(), userRequestDTO.getPassword());
+    }
+
+    @GetMapping("/all")
```
Rota de getAll é apenas "usuarios", não precisa adicionar nada aqui

MP-David committed:
- Estrutura arquitetura de packges
- Criação do Spring SecurityConfig
- Criação do JwtAuthenticationFilter
- Crud User
- Adicionado Mapper DTO
- Adicionado Properties de banco de JWT

MP-David committed:
Merge branch 'develop' into feature/config-jwt

# Conflicts:
#	backend/.gitignore
#	backend/pom.xml
#	backend/src/main/resources/application.properties
#	backend/target/classes/application.properties
